compiler/rustc_trait_selection/src/solve/trait_goals.rs

```diff
@@ -89,6 +89,15 @@ impl<'tcx> assembly::GoalKind<'tcx> for TraitPredicate<'tcx> {
         ecx: &mut EvalCtxt<'_, 'tcx>,
         goal: Goal<'tcx, Self>,
     ) -> QueryResult<'tcx> {
+        if let Some(def_id) = ecx.tcx().find_map_relevant_impl(
```

Comment:
```suggestion
        // This differs from the current stable behavior and
        // fixes #84857. Due to breakage found via crater, we
        // currently instead lint patterns which can be used to
        // exploit this unsoundness on stable, see #93367 for
        // more details.
        if let Some(def_id) = ecx.tcx().find_map_relevant_impl(
```
context is good i think

Reply:
Whoops, yeah, sorry.